src/pages/Login/Login.test.tsx

```diff
@@ -16,43 +16,46 @@ test("renders a login button", () => {
   expect(container.querySelector("ion-button")).toHaveTextContent(/login/i);
 });
 
-test("clicking the login button logs user in", () => {
-  const mockSetIsAuthenticated = jest.fn();
-  const { container } = render(
-    <AuthContext.Provider
-      value={{
-        isAuthenticated: false,
-        setIsAuthenticated: mockSetIsAuthenticated,
-      }}
-    >
+describe("Clicking login button", () => {
+  it("should call the login function", () => {
+    const mockLogin = jest.fn();
+    const mockUseAuth = jest.spyOn(UseAuthModule, "default").mockReturnValue({
+      isAuthenticated: false,
+      login: mockLogin,
+      logout: jest.fn(),
+    });
+
+    const { container } = render(
+      <AuthProvider>
+        <MemoryRouter initialEntries={["/login"]}>
+          <Login />
+        </MemoryRouter>
+      </AuthProvider>
+    );
+
+    userEvent.click(container.querySelector("ion-button") as TargetElement);
+
+    expect(mockLogin).toHaveBeenCalledTimes(1);
+
+    mockUseAuth.mockRestore();
+  });
+
+  it("should redirect user to home", () => {
```

Comment:
Note: may or may not be true. In the case of deep links the user would not be redirected to home.
Sure you thought of it, just bringing it up just in case